preload/link-header-preload-non-html.html

```diff
@@ -0,0 +1,38 @@
+<!DOCTYPE html>
+<meta charset=utf-8>
+<title>Makes sure that Link headers preload resources</title>
+<meta name="timeout" content="long">
+<script src="resources/dummy.js?link-header-preload2"></script>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="/preload/resources/preload_helper.js"></script>
+<body>
+<script>
+    const preloadLink = '/preload/resources/dummy.css';
+    function url_with_preload(content, type) {
+        const link = encodeURIComponent(`<${preloadLink}>;rel=preload;as=style`)
+        return getAbsoluteURL(`./resources/echo-preload-header.py?type=${type}&content=${encodeURIComponent(content)}&link=${link}`);
+    }
+
+    function test_document_type(content, type) {
+        promise_test(async t => {
+            const url = url_with_preload(content, type);
+            const win = window.open(url);
+            t.add_cleanup(() => win.close());
+            await new Promise(resolve => win.addEventListener('load', resolve));
+            const link = win.document.createElement('link');
+            link.rel = 'preload';
```

Comment:
Why are you adding a preload link here? Aren't you aiming to test the header preload?